src/Microsoft.VisualStudio.Services.Agent/Blob/DedupManifestArtifactClientFactory.cs

```diff
@@ -122,8 +132,15 @@ private DedupManifestArtifactClientFactory()
                 continueOnCapturedContext: false);
 
             var telemetry = new BlobStoreClientTelemetry(tracer, dedupStoreHttpClient.BaseAddress);
-            var client = new DedupStoreClientWithDataport(dedupStoreHttpClient, maxParallelism);
-            return (new DedupManifestArtifactClient(telemetry, client, tracer), telemetry);
+            traceOutput($"Hashtype: {this.HashType.Value}");
+
+            if (this.HashType == BuildXL.Cache.ContentStore.Hashing.HashType.Dedup1024K)
```

Comment:
>HashType [](http://example.com/codeflow?start=21&length=8)

hashtype could be a local (not blocking)

Reply:
I am going to punt the change. The idea is to have the same HashType for the singleton instance. Doesn't mean it can't be local.